reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt

```diff
@@ -0,0 +1,241 @@
+
+            val outputFile = outputDir.resolve("${filePrefix}${file.nameWithoutExtension}")
+            outputFiles += outputFile
+
+            freemarkerConfig.setDirectoryForTemplateLoading(file.parentFile)
+
+            val template = freemarkerConfig.getTemplate(file.name)
+            outputFile.writer().use { template.process(dataModel, it) }
+        }
+
+        return outputFiles
+    }
+
+    /**
+     * License information for a single package or project.
+     */
+    class PackageModel(
+        val id: Identifier,
+        private val input: ReporterInput
+    ) {
+        /**
+         * True if the package is excluded.
+         */
+        val excluded: Boolean by lazy { input.ortResult.isExcluded(id) }
+
+        /**
+         * The resolved license information for the package.
+         */
+        val license: ResolvedLicenseInfo by lazy {
+            val resolved = input.licenseInfoResolver.resolveLicenseInfo(id)
+            resolved.copy(licenses = resolved.licenses.sortedBy { it.license.toString() })
+        }
+
+        /**
+         * The resolved license file information for the package.
+         */
+        val licenseFiles: ResolvedLicenseFileInfo by lazy { input.licenseInfoResolver.resolveLicenseFiles(id) }
+
+        /**
+         * Return all [ResolvedLicense]s for this package excluding those licenses which are contained in any of the
+         * license files. This is useful when the raw texts of the license files are included in the generated output
+         * file and all licenses not contained in those files shall be listed separately.
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun licensesNotInLicenseFiles(): List<ResolvedLicense> {
+            val outputFileLicenses = licenseFiles.files.flatMap { it.licenses }
+            return license.filter { it !in outputFileLicenses }
+        }
+    }
+
+    /**
+     * A collection of helper functions for the Freemarker templates.
+     */
+    class TemplateHelper(private val ortResult: OrtResult, private val licenseConfiguration: LicenseConfiguration) {
+        /**
+         * Return [packages] that are a dependency of at least one of the provided [projects][projectIds].
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun filterByProjects(
+            packages: Collection<PackageModel>,
+            projectIds: Collection<Identifier>
+        ): List<PackageModel> {
+            val dependencies = projectIds.mapNotNull { ortResult.getProject(it) }
+                .flatMapTo(mutableSetOf()) { it.collectDependencies() }
+
+            return packages.filter { pkg -> pkg.id in dependencies }
+        }
+
+        /**
+         * Return licenses that are configured to require a
+         * [source code offer in a notice file][License.includeSourceCodeOfferInNoticeFile] in the
+         * [LicenseConfiguration].
+         */
```

Comment:
These docs are not valid anymore (and also not present anymore in the original `NoticeTemplateReporter.kt`), so we should just drop them.